app/scripts/modules/core/src/pipeline/config/stages/stage.module.js

```diff
@@ -73,6 +73,20 @@ module.exports = angular.module('spinnaker.core.pipeline.config.stage', [
         requisiteStageRefIds.includes(stage.refId) ? null : 'Downstream dependencies (unavailable)';
     };
 
+    $scope.stageProducesArtifacts = function() {
+      if (!$scope.stage) {
+        return false;
+      }
+
+      const stageConfig = pipelineConfig.getStageConfig($scope.stage);
+
+      if (!stageConfig) {
+        return false;
+      } else {
+        return stageConfig.producesArtifacts;
```

Comment:
consider `return !!stageConfig.producesArtifacts;` here so that `undefined` can't sneak in to the implicit function signature (will occur when a stage's config doesn't have a `producesArtifacts` field).